cogs/detect_code.py

```diff
@@ -71,15 +74,25 @@ def likely_contains_code(text: str) -> bool:
             return False
 
         non_blank_lines = [line for line in lines if line.strip()]
+
+        patterns = [
+            r"\b\w+\s*\(\s*.*?\s*\)\s*",
+            r"(#.*$)",
+            r"'''",
+            r'"""',
+            r";(?=\s*(//|#).+|$)",
+            r"^[\w]+\.[\w]+(\.[\w]+)*$",
+            r"\b\w+(?:_\w+)*\b(?:->\b\w+(?:_\w+)*\b)+",
+            r"^\/\*",
+            r".*\*/$",
+            r"[A-Za-z]+(?:[a-z][A-Z]|[A-Z][a-z])[A-Za-z]*",
+            r"^\w+(?:_\w+)+$",
+            r"\b[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)+\b",
+        ]
+        combined_pattern = "|".join(patterns)
```

Comment:
Combining the patterns like this scares me - it works, but in the future people may try to add new patterns which _are not_ compatible with this pipe join. I would just leave them separate, and later do `if any(re.search(line, pattern) for pattern in patterns)`

Reply:
Sure, I'll do that.